src/averageAges.js

```diff
@@ -20,6 +20,19 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const eligibleMen = !century
+    ? people.filter(person => person.sex === 'm')
+    : people.filter(person => {
+      const diedCentury = Math.ceil(person.died / 100);
+
+      return person.sex === 'm' && diedCentury === century;
+    });
+
+  const eligibleMenCount = eligibleMen.length;
+
+  return eligibleMen
+    .reduce((sum, man) => sum + (man.died - man.born), 0)
+       / eligibleMenCount;
```

Comment:
you do the same on calculateWomenAverageAge. You can create separate function and you it here and on calculateWomenAverageAge